tests/build.go

```diff
@@ -432,3 +434,66 @@ RUN ping -c 1 buildx.host-gateway-ip.local
 		require.NoError(t, err, string(out))
 	}
 }
+
+func testBuildNetworkModeBridge(t *testing.T, sb integration.Sandbox) {
+	if sb.Name() != "docker" {
+		t.Skip("skipping test for non-docker workers")
+	}
+
+	var builders []string
+	t.Cleanup(func() {
+		for _, builder := range builders {
+			out, err := rmCmd(sb, withArgs(builder))
+			require.NoError(t, err, out)
+		}
+	})
+
+	dockerfile := []byte(`
+FROM busybox AS build
+RUN ip a show eth0 | awk '/inet / {split($2, a, "/"); print a[1]}' > /ip.txt
+FROM scratch
+COPY --from=build /ip.txt /`)
+	dir := tmpdir(t, fstest.CreateFile("Dockerfile", dockerfile, 0600))
+
+	cases := []struct {
+		name string
+		args []string
+	}{
+		{
+			name: "bridge",
+			// TODO: use stable buildkit image when v0.13.0 released
+			args: []string{"--driver", "docker-container", "--buildkitd-flags=--oci-worker-net=bridge", "--driver-opt", "image=moby/buildkit:master"},
+		},
+		{
+			name: "host",
+			args: []string{"--driver", "docker-container", "--buildkitd-flags=--oci-worker-net=host"},
+		},
+	}
+
+	for _, tt := range cases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			out, err := createCmd(sb, withArgs(tt.args...))
+			require.NoError(t, err, out)
+			builderName := strings.TrimSpace(out)
+			builders = append(builders, builderName)
+
+			cmd := buildxCmd(sb, withArgs("build", fmt.Sprintf("--output=type=local,dest=%s", dir), dir))
+			cmd.Env = append(cmd.Env, "BUILDX_BUILDER="+builderName)
+			outb, err := cmd.CombinedOutput()
+			require.NoError(t, err, string(outb))
+
+			dt, err := os.ReadFile(filepath.Join(dir, "ip.txt"))
+			require.NoError(t, err)
+
+			ip := net.ParseIP(strings.TrimSpace(string(dt)))
+			require.NotNil(t, ip)
+
+			if tt.name == "bridge" {
+				_, subnet, err := net.ParseCIDR("10.10.0.0/16") // default subnet for bridge network in buildkit: https://github.com/moby/buildkit/blob/489f5fc6a331530daa8a31fb6ed7ba223df8ecb9/cmd/buildkitd/main.go#L474
+				require.NoError(t, err)
+				require.True(t, subnet.Contains(ip))
+			}
+		})
+	}
```

Comment:
@tonistiigi I use this kind of test cases to check the behavior to avoid this issue encountered in https://github.com/docker/buildx/pull/2270#discussion_r1498853203

Reply:
I guess this should be fine but the case I thought is to compare the values of host and bridge, to make sure bridge was applied and we don't get the default host even when the request did not set `--network`.

So instead of two buildkitd's with different network config, only one buildkitd with bridge mode and then test that in this buildkitd with `--network host` and without it provides a different ip/interfaces.

Reply:
Updated test